### PR TITLE
(maint) add linting, update dependencies, fix reflection issues 

### PR DESCRIPTION
**Pull request by jonathannewman**
This updates dependencies, adds in eastwood and clj-kondo as github
actions, addresses the issues found by clj-kondo and eastwood.

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -2,21 +2,20 @@
   :description "Clojure ldap client (Puppet Labs's fork)."
   :url "https://github.com/puppetlabs/clj-ldap"
   :dependencies [[org.clojure/clojure "1.10.1"]
-                 [com.unboundid/unboundid-ldapsdk "6.0.7"]]
+                 [com.unboundid/unboundid-ldapsdk "6.0.10"]]
   :source-paths ["src/clojure"]
   :java-source-paths ["src/java"]
-  :profiles {:dev {:dependencies [[jline "0.9.94"]
```
Comment (jonathannewman) [inline project.clj:8]: This wasn't being used at all.

File: project.clj [hunk 2]
```diff
@@ -2,21 +2,20 @@
...
-                                  [org.apache.directory.server/apacheds-all "1.5.7"
+  :profiles {:dev {:dependencies [[org.apache.directory.server/apacheds-all "1.5.7"
                                    ;; This dependency causes the classpath to contain two copies of the schema,
                                    ;; which prevents the test Directory Service from starting
                                    :exclusions [org.apache.directory.shared/shared-ldap-schema]]
-                                  [fs "1.1.2"]
```
Comment (jonathannewman) [inline project.clj:13]: This dependency was removed in favor of the java file IO layer

**Review by rileynewton — APPROVED** [review]
👍

**Review by artlawson — APPROVED** [review]
LGTM!

**Pull request merged**